apps/vscode/vscode.py

```diff
@@ -112,12 +112,31 @@ def line_swap_down():
     def line_clone():
         actions.key("shift-alt-down")
 
+    def line_insert_down(): # slap
+        actions.edit.line_end()
+        actions.key("escape")  # close completion popup
+        actions.key("enter")
+
     def jump_line(n: int):
         actions.user.vscode("workbench.action.gotoLine")
         actions.insert(str(n))
         actions.key("enter")
         actions.edit.line_start()
 
+    def selected_text() -> str:
+        with clip.capture() as s:
+            actions.edit.copy()
+        try:
+            txt = s.text()
+        except clip.NoChange:
+            return ""
+        if txt and txt[-1] == '\n' and sum(1 for c in txt if c == '\n') == 1 and txt not in ('\n', '\r\n'):
```

Comment:
Won't this break if someone has actually selected a line to copy it?

Reply:
Yes, it's a heuristic.
I think it's rare to manually copy a full line including the line terminator -- e.g. `take line` doesn't include the terminator. And a manual selection with `shift-down` will tend to include the indentation of the following line. So I think it's more likely that the selection was empty and the user didn't turn off the VSCode setting, than that the user really wanted this type of selection.

Also note that this is just about the parts of talon that care about `selected_text` -- the actual copy action will still work even if this heuristic was wrong.

Reply:
Mixed feelings. I agree that it will be vastly more common for this heuristic to help than hurt, but when it does hurt, it will be quite surprising

We could do this using a VSCode action if we added one to the talon extension, but probably good to have a fallback even so

Any thoughts @knausj85 ?

Reply:
I'm a bit hesitant since eg selecting lines via the line numbers does include the terminator, so this could break kinda often depending on folks workflow (eg eye tracking, some folks can use a mouse)

I'm on the fence atm

Reply:
I would be inclined to leave this part out, and instead add something to a VSCode extension, eg the Talon VSCode extension.  

What might make sense is to add something to the [knausj VSCode README](https://github.com/knausj85/knausj_talon/blob/master/apps/vscode/README.md) suggesting to turn off the setting to copy entire line, and if they don't, put the code snippet in there, or maybe link to something in the wiki

Reply:
That sounds reasonable to me!

Reply:
I'm going to remove this controversial bit from the PR and then merge the rest of it.